src/routeProposal/filtering.ts

```diff
@@ -39,8 +41,9 @@ export function filterPoolsOfInterest(
     tokenIn: string,
     tokenOut: string,
     maxPools: number,
+    chainId: number,
     currentBlockTimestamp = 0
-): [PoolDictionary, string[]] {
+): [PoolDictionary, string[], StablePool] {
```

Comment:
I'm not a fan of functions which return many different things. Can we pull this StablePool out elsewhere?